### PR TITLE
Add Truffleruby support

### PR DESCRIPTION
**Pull request by gogainda**
Add Truffleruby support to CI
Extract Rubocop linting into separate workflow

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
Out of curiosity: why did you extract rubocop from the default rake tasks?

I think it is a good idea indeed. But will that have any impact on the Truffleruby support?

**Comment by gogainda** [comment]
another example of the memory related issue what team is currently investigating: https://github.com/oracle/truffleruby/issues/2591

**Comment by gogainda** [comment]
@aslakhellesoy @aurelien-reeves is there anything else I can do here?

File: Rakefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ RuboCop::RakeTask.new
 require 'cucumber/rake/task'
 Cucumber::Rake::Task.new
 
-default_tasks = %i[spec cucumber cck rubocop]
```
Comment (luke-hill) [inline Rakefile:15]: Is this because truffleruby won't run rubocop? Or some other reason?

Comment (gogainda) [reply Rakefile:15]: > The main reason is that currently we have memory issue with GithubActions and I noticed that running tests together with rubocop fails the pipeline, which is currently the team is investigating, but I didn't want to block this PR with that Regarding the Truffleruby support, if pipeline is failing and others no please create an issue in the Truffleruby repository and the team will help with that

**Comment by aurelien-reeves** [comment]
> @aslakhellesoy @aurelien-reeves is there anything else I can do here?

Looks pretty good actually, thank you for your contribution.

Would you have the possibility to update the README to add truffleruby in the supported platforms? And the CHANGELOG too?

**Comment by gogainda** [comment]
@aurelien-reeves done

**Comment by eregon** [comment]
oO I missed that truffleruby is passing all cucumber tests now, that's great :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Please visit [cucumber/CONTRIBUTING.md](https://github.com/cucumber/cucumber/blo
 ## [Unreleased](https://github.com/cucumber/cucumber-ruby/compare/v8.0.0.rc.1...main)
 
 ### Added
+- Add support for TrufflrRuby ([PR#1612](https://github.com/cucumber/cucumber-ruby/pull/1612) [gogainda](https://github.com/gogainda))
```
Comment (eregon) [inline CHANGELOG.md:16]: TruffleRuby* and I think the entry should be similar to the ones below (see line 39 for example), so empty line before, PR/author link on a separate line, etc

**Comment by gogainda** [comment]
@eregon thanks for the review!

**Comment by aslakhellesoy** [comment]
Hi @gogainda,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by eregon** [comment]
The error is:
```
 rake cucumber
Running examples/sinatra
/home/runner/.rubies/truffleruby-head/bin/truffleruby -I "lib" "/home/runner/work/cucumber-ruby/cucumber-ruby/bin/cucumber" --format pretty
Feature: Addition
  In order to avoid silly mistakes
  As a math idiot 
  I want to be told the sum of two numbers

  Scenario: Add two numbers           # features/add.feature:6
    Given I visit the calculator page # features/step_definitions/add_steps.rb:1
      TruffleRuby doesn't have a case for the org.truffleruby.language.objects.IsImmutableObjectNodeGen node with values of type com.oracle.truffle.llvm.runtime.except.LLVMPolyglotException
      	from org.truffleruby.language.objects.IsImmutableObjectNodeGen.executeAndSpecialize(IsImmutableObjectNodeGen.java:117)
      	from org.truffleruby.language.objects.IsImmutableObjectNodeGen.execute(IsImmutableObjectNodeGen.java:59)
      	from org.truffleruby.core.kernel.KernelNodes$DupNode.dup(KernelNodes.java:685)
      	from org.truffleruby.core.kernel.KernelNodesFactory$DupNodeFactory$DupNodeGen.executeAndSpecialize(KernelNodesFactory.java:2232)
      	from org.truffleruby.core.kernel.KernelNodesFactory$DupNodeFactory$DupNodeGen.execute(KernelNodesFactory.java:2213)
      	from org.truffleruby.language.methods.CallInternalMethodNode.alwaysInlined(CallInternalMethodNode.java:114)
      	from org.truffleruby.language.methods.CallInternalMethodNodeGen.executeAndSpecialize(CallInternalMethodNodeGen.java:199)
      	from org.truffleruby.language.methods.CallInternalMethodNodeGen.execute(CallInternalMethodNodeGen.java:87)
      	from org.truffleruby.language.dispatch.DispatchNode.dispatch(DispatchNode.java:308)
      	from org.truffleruby.language.dispatch.RubyCallNode.doCall(RubyCallNode.java:152)
      	from org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:109)
      	from org.truffleruby.language.dispatch.RubyCallNode.executeArguments(RubyCallNode.java:185)
      	from org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:98)
      	from org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:89)
      	from org.truffleruby.language.dispatch.RubyCallNode.executeArguments(RubyCallNode.java:185)
      	from org.truffleruby.language.dispatch.RubyCallNode.execute(RubyCallNode.java:98)
      	from org.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:37)
      	from org.truffleruby.language.RubyMethodRootNode.execute(RubyMethodRootNode.java:65) (TypeError)
```

I'll take a look (cc @bjfish).

**Comment by aurelien-reeves** [comment]
Thanks @eregon :)

**Comment by eregon** [comment]
Indeed, it's green now: https://github.com/cucumber/cucumber-ruby/runs/6382227723?check_suite_focus=true

**Comment by eregon** [comment]
Yes, feel free to try `truffleruby` instead/additionally which uses the latest release.

Testing against `truffleruby-head` is helpful for the TruffleRuby team (and we monitor such failures via https://github.com/eregon/truffleruby-gem-tracker/, we noticed it just a bit after the ping), but we also understand if you want more stability.

Testing both also makes sense, then it's easy to differentiate if some change in cucumber/deps broke on TruffleRuby in general or if it's due to some change in truffleruby-head.

**Comment by aurelien-reeves** [comment]
> Yes, feel free to try `truffleruby` instead/additionally which uses the latest release.
> 
> Testing against `truffleruby-head` is helpful for the TruffleRuby team (and we monitor such failures via https://github.com/eregon/truffleruby-gem-tracker/, we noticed it just a bit after the ping), but we also understand if you want more stability.
> 
> Testing both also makes sense, then it's easy to differentiate if some change in cucumber/deps broke on TruffleRuby in general or if it's due to some change in truffleruby-head.

Noted! Thanks for the info! :)
We'll see what we could do

**Pull request merged**